en/advanced/parameters_and_configurations.md

```diff
@@ -330,6 +333,26 @@ The purpose of each line is given below (for more detail see [module_schema.yaml
  */
 ```
 
+#### Publishing Parameter Metadata to a GCS
+
+Parameter metadata is collected into a JSON or XML file during each PX4 build.
+For all targets with enough FLASH available the JSON file is xz-compressed and stored within the generated binary.
+The file is then shared to ground stations using the [MAVLink Component Information Protocol](https://mavlink.io/en/services/component_information.html).
+This ensures that parameter metadata is always up-to-date with the code running on the vehicle.
+
+Binaries for flight controller targets with constrained memory (`CONSTRAINED_MEMORY`) do not store the *build-specific* parameter metadata in the binary, but instead reference metadata stored on `px4-travis.s3.amazonaws.com`.
+Versions of the metadata exist for the current master and each PX4 release.
+The metadata is uploaded via [github CI](https://github.com/PX4/PX4-Autopilot/blob/master/.github/workflows/metadata.yml) (and hence will only be available once parameters have been merged into master).
+
+:::note
+The metadata on `px4-travis.s3.amazonaws.com` is used as a fallback if parameter metadata is not present on the vehicle.
+It may also be used as a fallback, for example, to avoid a very slow download over a low-rate telemetry link.
+:::
+
+Anyone doing custom development on a FLASH-constrained board can adjust the URL [here](https://github.com/PX4/PX4-Autopilot/blob/master/src/lib/component_information/CMakeLists.txt#L41) to point to another server.
+
+The XML file of the master branch is copied into the QGC source tree via CI and is used as a fallback in cases where no metadata is available via the component information service (this approach predates the existence of the component information protocol).
```

Comment:
So this still happens right, and is the only use of the XML files?

Reply:
Yes still in use. At some point we can disable that.
Currently also Flight Review uses the xml. I might change that but don't see immediate need.